File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

unsigned long square_of_sum(const unsigned long n) {
  unsigned long sum = 0;
  for (unsigned long i = 1; i <= n; i++) {
    sum += i;
  }
  return sum * sum;
}

unsigned long sum_of_squares(const unsigned long n) {
  unsigned long sum = 0;
  for (unsigned long i = 1; i <= n; i++) {
    sum += i * i;
  }
  return sum;
}

unsigned long difference_of_squares(const unsigned long n) {
  return square_of_sum(n) - sum_of_squares(n);
}

File: javascript/difference-of-squares/difference-of-squares.js
Array.range = function(max) {
    return [...Array(max).keys()].map(x => x + 1);
}

Array.prototype.sum = function() {
    return this.reduce((x, y) => x + y);
}

Array.prototype.square = function() {
    return this.map(x => x * x);
}

export class Squares {
    constructor(max) {
        this.squareOfSum = Array.range(max).sum() ** 2;
        this.sumOfSquares = Array.range(max).square().sum();
        this.difference = this.squareOfSum - this.sumOfSquares;
    }
}
